Core/Src/obc_handler.c

```diff
@@ -0,0 +1,251 @@
+#include "command_handler.h"
+#include "iris_system.h"
+#include "arducam.h"
+#include "obc_handler.h"
+#include "debug.h"
+#include "spi_obc.h"
+#include "iris_time.h"
+#include "spi_bitbang.h"
+
+extern SPI_HandleTypeDef hspi1;
+extern uint8_t cam_to_nand_transfer_flag;
+
+static uint32_t count = 0x0FFF0000;
+uint8_t sensor_mode = 0;
+
+void transfer_image_to_obc();
+int step_transfer();
+
+const uint8_t iris_commands[IRIS_NUM_COMMANDS] = {IRIS_TAKE_PIC,
+                                                  IRIS_GET_IMAGE_LENGTH,
+                                                  IRIS_TRANSFER_IMAGE,
+                                                  IRIS_GET_IMAGE_COUNT,
+                                                  IRIS_ON_SENSORS,
+                                                  IRIS_OFF_SENSORS,
+                                                  IRIS_SEND_HOUSEKEEPING,
+                                                  IRIS_UPDATE_SENSOR_I2C_REG,
+                                                  IRIS_UPDATE_CURRENT_LIMIT,
+                                                  IRIS_SET_TIME,
+                                                  IRIS_WDT_CHECK};
+
+/*
+ * Refactor Iris
+ * - No processing of command occurs, delegate it to command_handler.h
+ * - spi drivers should be placed in the driver file **
+ * - Only verification and facilitation of commands should happen hear
+ */
+
+/**
+ * @brief
+ * 		Verifies if command from OBC is valid or not
+ * @param
+ * 		obc_cmd: Command from OBC
+ * @return
+ * 		1 if valid command, 0 if not
+ */
+int obc_verify_command(uint8_t obc_cmd) {
+    uint8_t ack = 0xAA;
+    uint8_t nack = 0x0F;
+    uint8_t transmit_ack;
+
+    transmit_ack = 0;
+
+    for (uint8_t index = 0; index < IRIS_NUM_COMMANDS; index++) {
+        if (iris_commands[index] == obc_cmd) {
+            transmit_ack = 1;
+        }
+    }
+
+    if (transmit_ack != 0) {
+        obc_spi_transmit(&ack, 2);
+        return 0;
+    } else {
+        obc_spi_transmit(&nack, 1);
+        return -1;
+    }
+}
+
+/**
+ * @brief
+ * 		Handles command from OBC
+ * @param
+ * 		obc_cmd: Command from OBC
+ * @return
+ * 		1 if valid command, 0 if not
+ */
+int obc_handle_command(uint8_t obc_cmd) {
+    uint8_t tx_data = 0x69;
```

Comment:
nice

Reply:
can probably get rid of this and replace with ack

Reply:
Done